sycl/source/handler.cpp

```diff
@@ -2535,6 +2535,20 @@ bool handler::HasAssociatedAccessor(detail::AccessorImplHost *Req,
 void handler::setType(sycl::detail::CGType Type) { impl->MCGType = Type; }
 sycl::detail::CGType handler::getType() const { return impl->MCGType; }
 
+void handler::setDeviceKernelInfo(kernel &&Kernel) {
+  MKernel = detail::getSyclObjImpl(std::move(Kernel));
+  MKernelName = getKernelName();
+  if (MKernel->isInterop()) {
+    // For regular kernels (created from user functors) the DeviceKernelInfo is
+    // set from the ProgramManager
```

Comment:
Is that even true? Isn't it done in https://github.com/intel/llvm/blob/cb03a1b8ab88e3aafc3060cb65fd22277d52991c/sycl/include/sycl/handler.hpp#L494-L502 ?